pkg/processor/k8sprocessor/kube/owner.go

```diff
@@ -338,7 +363,13 @@ func (op *OwnerCache) addOwnerInformer(
 		}),
 	})
 	if err != nil {
-		op.logger.Error("error adding event handler to namespace informer", zap.Error(err))
+		op.logger.Error("error adding event handler to owner informer", zap.Error(err), zap.String("kind", kind))
+	}
+
+	if transformFunc != nil {
+		if err = informer.SetTransform(transformFunc); err != nil {
```

Comment:
Is the transform function going to remove stale cache entries?

Reply:
No, this doesn't fix the memory leak at all. I added a test for it, but it's skipped for now.